testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveViews.java

```diff
@@ -172,5 +180,53 @@ public void testLateralViewJsonTupleAs()
                 "SELECT * FROM test_json_tuple_view",
                 queryAssert -> queryAssert.containsOnly(row(3, "Mateusz", "Gajewski", "true", "1000", null, null)));
     }
+
+    @Test(groups = HIVE_VIEWS)
+    public void testUnionViews()
+    {
+        onHive().executeQuery("DROP VIEW IF EXISTS union_all_view");
+        onHive().executeQuery("DROP VIEW IF EXISTS union_view");
+        onHive().executeQuery("DROP VIEW IF EXISTS union_distinct_view");
+
+        Map<String, List<QueryAssert.Row>> expected = new HashMap<>();
+
+        // UNION ALL
+        onHive().executeQuery("CREATE VIEW union_all_view AS\n"
+                + "SELECT n_regionkey FROM nation\n"
+                + "UNION ALL\n"
+                + "SELECT n_regionkey FROM nation\n");
+        expected.put(
+                "union_all_view",
+                // 10 copies of each number from 0 to 4
+                nCopies(10, List.of(row(0), row(1), row(2), row(3), row(4))).stream()
+                        .flatMap(List::stream)
+                        .collect(toList()));
+
+        // UNION DISTINCT
+        onHive().executeQuery("CREATE VIEW union_distinct_view AS\n"
+                + "SELECT n_regionkey FROM nation\n"
+                + "UNION DISTINCT\n"
+                + "SELECT n_regionkey FROM nation\n");
+        expected.put(
+                "union_distinct_view",
+                List.of(row(0), row(1), row(2), row(3), row(4)));
+
+        // UNION
+        // add 1 so neither side of the union is a subset of the other
+        onHive().executeQuery("CREATE VIEW union_view AS\n"
+                + "SELECT 1 + n_regionkey FROM nation\n"
+                + "UNION\n"
```

Comment:
since UNION = UNION DISTINCT, let's have it as 

```
for (String unionDistinct = List.of("UNION", "UNION DISTINCT")) {
```

so that the code is terser and it's very clear they are tested the same way

Reply:
Heh, I had that originally, but thought it might be too fancy. I'll put it back.